lib/trino-filesystem/src/test/java/io/trino/filesystem/AbstractTestTrinoFileSystem.java

```diff
@@ -1003,7 +1003,8 @@ private static String retrieveUri(UriLocation uriLocation)
             try {
                 HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
                 if (response.statusCode() != 200) {
-                    throw new IOException("Failed to retrieve");
+                    throw new IOException("Failed to retrieve, got response code: %d, body: %s"
```

Comment:
is the body readable (non-binary)?

Reply:
yes, the error messages are usually xml/text